### ISSUE TITLE
Is it fine not use Softmax after forwarding?

### ISSUE THREAD

**Issue by graykode** [openai/gpt-2#51]
@WuTheFWasThat 
![image](https://user-images.githubusercontent.com/10525011/53002114-4549f600-346f-11e9-9f89-bc4c7debd569.png)
I found something different from the paper.
https://github.com/openai/gpt-2/blob/master/src/model.py#L171
It may be mabe some error when `tf.multinomial(logits, num_samples=1, output_dtype=tf.int32)` function
because there will be value less than zero
I try with softmax but it does not shows good result for text-generator..
This is result when use softmax function
![image](https://user-images.githubusercontent.com/10525011/53003402-39136800-3472-11e9-8e68-fd7d39cd80f8.png)
Not using softmax is your original meant?

I added softmax before multinomial sampling when implement to Pytorch to avoid encountering probability entry < 0
https://github.com/graykode/gpt-2-Pytorch/blob/master/sample.py#L43

**Comment by graykode** [comment]
Also Does `n_ctx` mean is `number of context` with same `k` where k is the size of the context window?? :D

**Comment by WuTheFWasThat** [comment]
tf.multinomial takes in logits, not probabilities

`n_ctx` is the size of the context window.  (the `k` in `top_k` refers to truncating the set of considered logits and is different.  not sure this was related to your confusion, but just in case)
